main.go

```diff
@@ -34,12 +35,21 @@ func main() {
 		logrus.WithError(err).Fatal("could not instantiate wireguard controller")
 	}
 
-	cluster, err := newCluster(config, wg)
+	getMeta := func(limit int) []byte {
```

Comment:
I disagree with implementing the interface. Pushing cluster-related concerns in the wireguard code and types is exactly what I would like to avoid. Implementing the interface in `wgState` would probably make the code a bit more concise, but it would complexify understanding, and then changes in `memberlist` will affect both the cluster *and* the wireguard code.

I intend (I have it working locally, just writing some more documentation before I commit) to move both wireguard and cluster code to separate module. It might be a bit more verbose, with additional `use` especially, but it is an assurance that we avoid mixing concerns too much: the glue code must be written in the main loop.

Reply:
I ended up making it an `Encode` function of the common `Node` type, which makes sense imo. And a `localNode` is passed to the cluster instance. The code is on a separate branch where I split the modules, so I'll push it later.

Reply:
ok, that sounds reasonable.

Could `Node` then implement `memberlist.Delegate`? My problem is that I never liked having the cluster implement `Delegate` itself. It was a lazy "solution" on my part.

Do you think we could clear this up before going for the split PR? Or are the changes too intertwined?

Reply:
Definitely yes. `Node` being a shared type, `cluster.go` can have it implement the delegate interface, that is probably the cleanest way to do it. I can try and change it before splitting, I'll resolve any conflict later on :)